identity_hasher.go

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package ionhash
+
+type identityHasher struct {
```

Comment:
Might be better to move code (and I understand this might be the majority of the implementation) into the `internal` folder. Go modules treats code inside an internal folder as private in the sense that only code located in the same parent path can access it. 

https://github.com/amzn/ion-hash-go/issues/25

Reply:
We will do it in a separate PR.